### PR TITLE
Braintree API

### PR DESCRIPTION
**Pull request by toddmedema**
Merging now so that I can betatest iOS and Android apps, for post-review

### CODE REVIEW

File: routes.js [hunk 1]
```diff
@@ -12,6 +13,13 @@ const Mail = require('./mail');
 const oauth2 = require('./lib/oauth2');
 const Quests = require('./models/quests');
 
+const gateway = Braintree.connect({
```
Comment (smartin015) [inline routes.js]: Prefer `braintree` or similar, since `gateway` is a tad generic.

File: routes.js [hunk 2]
```diff
@@ -195,4 +203,37 @@ router.post('/user/subscribe', (req, res) => {
...
+    res.send(response.clientToken);
+  });
+});
+
+
+router.post('/braintree/checkout', (req, res) => {
+  req.body = JSON.parse(req.body);
+  res.header('Access-Control-Allow-Origin', req.get('origin'));
+  res.header('Access-Control-Allow-Credentials', 'true');
+  gateway.transaction.sale({
```
Comment (smartin015) [inline routes.js]: How will this record the user requesting the payment?

File: routes.js (hunk 1 above)
Comment (toddmedema) [inline routes.js]: Good point

File: routes.js (hunk 2 above)
Comment (toddmedema) [inline routes.js]: Ah, yep, just updated to capture customer and product info :)

**Pull request merged**